src/workloads/execution/TimeSeriesSort.yml

```diff
@@ -0,0 +1,512 @@
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 103680
+    BatchSize: *batchSize
+    Document:
+      t: {^IncDate: {start: "2022-01-01", step: 400}}
+      m: {^RandomInt: {min: 0, max: 1000}}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 5000
+    BatchSize: *batchSize
+    Document:
+      t: {^IncDate: {start: "2022-01-01", step: 400}}
+      m: 1001
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10
+    BatchSize: *batchSize
+    Document:
+      t: {^IncDate: {start: "2024-01-01", step: 400}}
+      m: 1002
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 503680
+    BatchSize: *batchSize
+    Document:
+      t: {^IncDate: {start: "2023-01-01", step: 400}}
+      m: {^RandomInt: {min: 0, max: 1000}}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Queries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: &numQueryRuns 50
+    Database: *db
+    Operations:
+    - OperationMetricsName: SortQueryBlockingSort
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *coll
+        pipeline: [{$sort: {t: 1}}]
+        cursor: {batchSize: *batchSize}
+  - Repeat: *numQueryRuns
+    Database: *db
+    Operations:
+    - OperationMetricsName: SortQueryBoundedSorter
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: system.buckets.Collection0
+        pipeline: [{$sort: {"control.min.t": 1}},
+                   {$_internalUnpackBucket: {
+                     exclude: [],
+                     timeField: "t",
+                     metaField: "m",
+                     bucketMaxSpanSeconds: 3600,
+                     assumeNoMixedSchemaData: true
+                   }
+                   },
+                   {$_internalBoundedSort: {
+                     sortKey: {t: 1},
+                     bound: 3600,
+                   }
+                   }]
+        cursor: {batchSize: *batchSize}
```

Comment:
I think this will only return the first batch of results.  I assume RunCommand doesn't have any special knowledge about cursors or getMore, so it would send this command as-is and wait for the response.

If that's right, we can force the whole query to run by adding some extra stage at the end.  Maybe {$skip: 1e10} or {$_internalInhibitOptimization: {}}, {$skip: 1e10}.  There must be examples in other workloads.

Reply:
Offline we decided on $skip for this one